### PR TITLE
CI: Notify slack on failure of the test-install workflow.

### PR DESCRIPTION
**Pull request by csadorf**

### CODE REVIEW

**Review by ltalirz — COMMENTED** [review]
thanks @csadorf - just two questions

 * does it work?
 * can the notification task fail? if yes, we probably want to ignore such a failure

**Comment by csadorf** [comment]
> thanks @csadorf - just two questions
> 
>     * does it work?

Yes, see #dev-aiida-core channel on the AiiDA slack workspace. 

> 
>     * can the notification task fail? if yes, we probably want to ignore such a failure

Yes, but I'd consider that an issue that I'd prefer not to silently ignore. The notification steps are currently only triggered on the "scheduled" runs. Since this would only be triggered if there is already an issue with the workflow, there is no benefit in silently ignoring this, is there?

File: .github/workflows/test-install.yml [hunk 1]
```diff
@@ -72,6 +72,14 @@ jobs:
       run:
         python -c "import aiida"
 
+    - name: Send Slack notification
+      if: ${{ failure() && github.event_name == 'schedule' }}
```
Comment (chrisjsewell) [inline .github/workflows/test-install.yml:76]: after #4689 is merged, I think this will never trigger, because the `continue-on-error` has been removed

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

File: .github/workflows/test-install.yml (hunk 1 above)
Comment (csadorf) [reply .github/workflows/test-install.yml:76]: The `continue-on-error` field was previously added to ensure that the overall workflow continues even if there are issues with the pip-2020-resolver. To the best of my knowledge this should not affect the execution of those steps to run on failure.

Comment (chrisjsewell) [reply .github/workflows/test-install.yml:76]: oh yeh I think you are correct

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
if you resolve the conflicts, I'm happy to approve 👍

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**